code/modules/admin/verbs/playsound.dm

```diff
@@ -174,6 +183,96 @@
+	if(!ytdl)
+		to_chat(src, span_boldwarning("Youtube-dl was not configured, action unavailable"), confidential = TRUE) //Check config.txt for the INVOKE_YOUTUBEDL value
+		return
+	var/web_sound_url
+	var/list/music_extra_data = list()
+	if(length(request))
+		request = trim(request)
+		if(findtext(request, ":") && !findtext(request, GLOB.is_http_protocol))
+			to_chat(src, span_boldwarning("Non-http(s) URIs are not allowed."), confidential = TRUE)
+			to_chat(src, span_warning("For youtube-dl shortcuts like ytsearch: please use the appropriate full url from the website."), confidential = TRUE)
+			return
+		var/shell_scrubbed_input = shell_url_scrub(replacetext(request, "music.", ""))
+		var/list/output = world.shelleo("[ytdl] --geo-bypass --format \"bestaudio\[ext=mp3]/best\[ext=mp4]\[height <= 360]/bestaudio\[ext=m4a]/bestaudio\[ext=aac]\" --dump-single-json --no-playlist -- \"[shell_scrubbed_input]\"")
+		var/errorlevel = output[SHELLEO_ERRORLEVEL]
+		var/stdout = output[SHELLEO_STDOUT]
+		var/stderr = output[SHELLEO_STDERR]
+		if(!errorlevel)
+			var/list/data
+			try
+				data = json_decode(stdout)
+			catch(var/exception/e)
+				to_chat(src, span_boldwarning("Youtube-dl JSON parsing FAILED:"), confidential = TRUE)
+				to_chat(src, span_warning("[e]: [stdout]"), confidential = TRUE)
+				return
+
+			if (data["url"])
+				web_sound_url = data["url"]
+				var/title = "[data["title"]]"
+				var/webpage_url = title
+				if (data["webpage_url"])
+					webpage_url = "<a href=\"[data["webpage_url"]]\">[title]</a>"
+				music_extra_data["duration"] = DisplayTimeText(data["duration"] * 1 SECONDS)
+				music_extra_data["link"] = data["webpage_url"]
+				music_extra_data["artist"] = data["artist"]
+				music_extra_data["upload_date"] = data["upload_date"]
+				music_extra_data["album"] = data["album"]
+
+				var/duration = null
+				if (data["duration"] * 1 SECONDS > 10 MINUTES)
+					duration = tgui_alert(usr, "This song is over 10 minutes long. Are you sure you want to play it?", "Length Warning!", list("No", "Yes", "Cancel"))
+				switch(duration)
+					if("No")
+						return
+					if("Cancel")
+						return
+
+				var/res = tgui_alert(usr, "Show the title of and link to this song to the players?\n[title]", "Show Info?", list("No", "Yes", "Cancel"))
+				switch(res)
+					if("Yes")
+						music_extra_data["title"] = data["title"]
+					if("No")
+						music_extra_data["link"] = "Song Link Hidden"
+						music_extra_data["title"] = "Song Title Hidden"
+						music_extra_data["artist"] = "Song Artist Hidden"
+						music_extra_data["upload_date"] = "Song Upload Date Hidden"
+						music_extra_data["album"] = "Song Album Hidden"
+					if("Cancel")
+						return
+
+				var/anon = tgui_alert(usr, "Display who played the song?", "Credit Yourself?", list("No", "Yes", "Cancel"))
+				switch(anon)
+					if("Yes")
+						if(res == "Yes")
+							to_chat(world, span_boldannounce("[src] played: [webpage_url]"), confidential = TRUE)
```

Comment:
in conjunction with the other changes, make this `[key_name(user)]` or something